### ISSUE TITLE
Infinite attempts to reload async relationship

### ISSUE THREAD

**Issue by AlexMayants** [emberjs/data#6467]
Suppose, we define two models, `book` and `author`, with `book` having `belongsTo` relationship with `author`. We can write a template like this

```hbs
Author: {{book.author.name}}
```

On access to the `author` property the model for the author will be loaded in background. However, if the network is down or the access to the resource is denied, the request fails. In this case, the `unloadRecord` fires on the `author`'s `internalModel`. It leads eventually to `notifyPropertyChange` on the `author` property of the `book`, which causes a new attempt to load the author. If the network problem persists or the user is unautorized, the loading attempts will repeat again and again.

I have observed this behavior in our application after upgrade to Ember 3.8.3 / Ember-Data 3.8.0. Here are the current versions:
```cli
├── ember-cli@3.8.3
├── ember-data@3.8.0
└── ember-source@3.8.3
```
It reproduces at least up to Ember 3.10 / Ember-Data 3.10.
```cli
+-- ember-cli@3.10.1
+-- ember-data@3.10.0
`-- ember-source@3.10.2
```

Unfortunately, we are unable to upgrade to Ember above 3.8 yet, because of some unsupported addons. Can this issue be fixed in the LTS or on the application side?

### Reproduction
Here's a codesandbox with a synthetic reproduction example: https://codesandbox.io/s/inifite-reload-g3oog

Try to visit Book 1 or Book 2 subpages. Notice how authors are loaded asynchronously.

Now block one of the `https://jsonapiplayground.reyesoft.com/v2/authors/XXX` URLs in the DevTools, refresh the test browser and visit the subpages again.

*The requests will be sent at a very high rate. Return to the root page and refresh the test browser before your web browser hangs.*

**Comment by Gaurav0** [comment]
Have you tried running Ember Data 3.12 with Ember 3.8?

**Comment by AlexMayants** [comment]
@Gaurav0, I haven't yet, but I'll try. Is this combination supported?

**Comment by Gaurav0** [comment]
It is tested as part of regular CI using ember-try, if that is what you are asking.

**Comment by AlexMayants** [comment]
@Gaurav0, it seems to work in our application, thanks!

**Comment by runspired** [comment]
Closing as a duplicate of #5814 which was resolved by #6112
